### PR TITLE
[Create Timepoint] Sanitize CandID in href

### PR DESCRIPTION
**Pull request by johnsaigle**
Redmine #13790

While CandID comes from the database and is expected to be trusted, this PR explicitly sanitizes the candID variable before it is injected into the href element.

### CODE REVIEW

**Comment by johnsaigle** [comment]
@driusan I wanted to get your eyes on this in particular as you created the Redmine ticket.

**Comment by johnsaigle** [comment]
Tagging this as security to match the Redmine ticket.

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 1]
```diff
@@ -81,6 +81,11 @@ class Create_Timepoint extends \NDB_Form
         );
 
         $this->tpl_data['success'] = true;
+        $this->tpl_data['candID']  = htmlentities(
```
Comment (zaliqarosli) [inline modules/create_timepoint/php/create_timepoint.class.inc]: why are we doing this twice, once in process() and once in setup() ?

Comment (johnsaigle) [reply modules/create_timepoint/php/create_timepoint.class.inc]: You're right it only needs to happen in setup. I'll change this in a bit.

Comment (johnsaigle) [reply modules/create_timepoint/php/create_timepoint.class.inc]: Actually looking at this again it looks like we need to do it twice... In both functions an element is added to the front-end and it needs to be sanitized there. 

From what I know of LORIS it seems like some functionality is duplicated here and I'm not sure that `_process` should be creating HTML elements. It should probably be refactored but that's not the goal of this PR so the sanitization should occur in both place.

Comment (zaliqarosli) [reply modules/create_timepoint/php/create_timepoint.class.inc]: but i believe that you added the addition of the element to the front-end in process(). like 84 to add the identifier to tpl_data wasn't there before. what was the reason for adding it?

Comment (johnsaigle) [reply modules/create_timepoint/php/create_timepoint.class.inc]: Excellent point... Sorry I was working on another PR when looking at this and was distracted. I'll remove it.

**Comment by johnsaigle** [comment]
@zaliqarosli I fixed it like you suggested.

**Comment by HenriRabalais** [comment]
@johnsaigle how should this be tested?

**Comment by johnsaigle** [comment]
@HenriRabalais If you look at the Redmine ticket there's some background there. Essentially Dave notes that we are protected "by accident" from XSS here due to some upstream quirk. This PR simply makes us extra sure. The functionality before and after this PR shouldn't change. However, as we restructure LORIS this existing accidental protection might fall away and this will stop us from having a security issue at that time.

In short, as long as this change doesn't break anything consider it to be passing.

**Comment by HenriRabalais** [comment]
Created a timepoint and accessed it. Everything appears to be in order!

**Comment by driusan** [comment]
I don't have any problems with this change, but it doesn't address the Redmine ticket that is referenced.

**Comment by driusan** [comment]
Moving this to minor because nothing's currently broken, so it's not worth risking breaking something in a bugfix release.

**Comment by johnsaigle** [comment]
@driusan From the ticket:

> I think we're safe entirely by accident, but **it still needs to be sanitized** and the JS should be updated.

This was just a quick fix to do sanitization until the module is converted to React.
